src/system/Card/Card.tsx

```diff
@@ -1,39 +1,64 @@
+/** @jsxImportSource theme-ui */
+
 /**
  * External dependencies
  */
 import classNames, { Argument } from 'classnames';
 import { forwardRef, Ref } from 'react';
-import { BoxProps } from 'theme-ui';
+import { BoxProps, useThemeUI, get } from 'theme-ui';
 
 /**
  * Internal dependencies
  */
 import { Box } from '..';
 
+export enum CardVariant {
+	'primary',
+	'secondary',
+	'indent',
+}
+
 export interface CardProps {
-	variant?: string;
+	variant?: keyof typeof CardVariant;
 	sx?: BoxProps[ 'sx' ];
 	className?: Argument;
+	header?: string;
+	children?: React.ReactNode;
 }
 
 type CardBoxProps = CardProps & BoxProps;
 
 export const Card = forwardRef< HTMLElement, CardBoxProps >(
 	(
-		{ variant = 'primary', sx = {}, className, ...props }: CardBoxProps,
+		{ variant = 'primary', header, sx = {}, className, children, ...props }: CardProps,
 		ref: Ref< HTMLElement >
-	) => (
-		<Box
-			ref={ ref }
-			sx={ {
-				// pass variant prop to sx
-				variant: `cards.${ variant }`,
-				...sx,
-			} }
-			className={ classNames( 'vip-card-component', className ) }
-			{ ...props }
-		/>
-	)
+	) => {
+		const { theme } = useThemeUI();
+		const cardVariant = get( theme, `cards.${ variant }`, {} );
+		const basePadding = cardVariant?.p || cardVariant?.padding || 3;
+		return (
+			<Box
+				ref={ ref }
+				sx={ {
+					// pass variant prop to sx
+					variant: `cards.${ variant }`,
+					p: header ? 0 : basePadding, // Remove padding if there is a header so it sits flush
+					...sx,
+				} }
+				className={ classNames( 'vip-card-component', className ) }
+				{ ...props }
+			>
+				{ header && <header aria-label="Card header">{ header }</header> }
```

Comment:
```suggestion
				{ header && <div>{ header }</header> }
```
While HTML5 allows the use of multiple `header` elements on a page, A11Y advises against it due to potential confusion for assistive technologies. 
Reference: https://dequeuniversity.com/rules/axe/4.2/landmark-no-duplicate-banner

The Aria role banner is similar to the header. You can use a div here; you don't need an aria-label. I think you should accept more than just a string on the header prop because this is a Heading in the end.